components/DonationAskModal.vue

```diff
@@ -66,106 +66,127 @@
 </template>
-<script>
-import { mapState } from 'pinia'
-import { useDonationStore } from '../stores/donations'
-import { useGroupStore } from '../stores/group'
-import DonationAskVideo from './DonationAskVideo'
-import DonationAskQuote from './DonationAskQuote'
-import DonationAskButtons2510 from './DonationAskButtons2510'
-import DonationAskButtons51025 from './DonationAskButtons51025'
-import modal from '@/mixins/modal'
-import api from '~/api'
-
-export default {
-  components: {
-    DonationAskVideo,
-    DonationAskQuote,
-    DonationAskButtons2510,
-    DonationAskButtons51025,
-  },
-  mixins: [modal],
-  props: {
-    groupid: {
-      type: Number,
-      required: false,
-      default: null,
-    },
-  },
-  async setup(props) {
-    const donationStore = useDonationStore()
-    const groupStore = useGroupStore()
-
-    if (props.groupid) {
-      await groupStore.fetchGroup(props.groupid)
-    }
 
-    const runtimeConfig = useRuntimeConfig()
+<script setup>
+import { storeToRefs } from 'pinia'
+import { useGroupStore } from '~/stores/group'
+import { useDonationStore } from '~/stores/donations'
+import { useMiscStore } from '~/stores/misc'
+import { useModal } from '~/composables/useModal'
+import { useRuntimeConfig } from '#app'
+import DonationAskVideo from '~/components/DonationAskVideo.vue'
+import DonationAskQuote from '~/components/DonationAskQuote.vue'
+import DonationAskButtons2510 from '~/components/DonationAskButtons2510.vue'
+import DonationAskButtons51025 from '~/components/DonationAskButtons51025.vue'
+import DonationAskWhatYouCan from '~/components/DonationAskWhatYouCan.vue'
+import Api from '~/api'
 
-    return {
-      donationStore,
-      groupStore,
-      api: api(runtimeConfig),
-    }
-  },
-  data() {
-    return {
-      variant: null,
-      monthly: false,
+const groupStore = useGroupStore()
+const donationStore = useDonationStore()
+const miscStore = useMiscStore()
+const runtimeConfig = useRuntimeConfig()
+const api = Api(runtimeConfig)
+
+const props = defineProps({
+  requestedVariant: { type: String, default: '' },
+})
+
+const emit = defineEmits(['hide'])
+
+const variant = ref(null)
+const { showModal, hide } = useModal(emit)
+
+const groupId = ref(null)
+
+const groupName = computed(() => {
+  if (groupId.value && !targetMet.value) {
+    return groupStore?.get(groupId.value)?.namedisplay
+  } else {
+    return 'Freegle'
+  }
+})
+
+const { raised, target } = storeToRefs(donationStore)
+
+const targetMet = computed(() => {
+  return groupId.value && raised.value > target.value
+})
+
+function ask() {
+  show(props.requestedVariant)
+
+  miscStore.set({
+    key: 'lastdonationask',
+    value: new Date().getTime(),
+  })
+}
+
+const lastAsk = miscStore.get('lastdonationask')
+const canAsk =
+  !lastAsk || new Date().getTime() - lastAsk > 60 * 60 * 1000 * 24 * 7
+
+// Donation ask on Browse page is only used when we have a specific push.
+// canAsk = false // Why is this?
+
+if (canAsk) {
```

Comment:
I think this means we'll ask whenever we visit the page, e.g. on My Posts.  That's useful for testing but we need to remove it, so that the ask only occurs when the outcome event triggers it.

Reply:
I don't understand. There were 2 triggers for the modal to be shown:

1. When there's an outcome for a post
2. Automatically, if the last time the modal was shown was more than 90 days ago.

Did I not understand the code correctly? I did not add any logic or modify the existing one, I simply transferred it from `myposts` to here.